src/common/Platform.h

```diff
@@ -63,15 +63,38 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define __x86_64__ 1
 #endif
 
-// SSE support
-#if defined(__x86_64__) || defined(__SSE__) || _M_IX86_FP >= 1
-#include <xmmintrin.h>
-#if defined(__x86_64__) || defined(__SSE2__) || _M_IX86_FP >= 2
-#include <emmintrin.h>
-#define idx86_sse 2
-#else
-#define idx86_sse 1
+/* The definition name syntax is: DAEMON_USE_INTRINSICS_<architecture>[_extension]
+
+Examples:
+
+- DAEMON_CPU_INTRINSICS_i686: i686 specific code, including asm code.
+- DAEMON_CPU_INTRINSICS_i686_sse: i686 SSE specific code.
+- DAEMON_CPU_INTRINSICS_i686_sse2: i686 SSE2 specific code.
+
+If a platform inherits a feature from an parent platform, the parent
+platform name is used. For example on amd64, the definition enabling
+SSE code is DAEMON_CPU_INTRINSICS_i686_sse, enabling SSE code on both
+i686 with SSE and amd64.
+
+The definitions for the CPU itself is automatically is set by CMake. */
+
+#if defined(DAEMON_USE_CPU_INTRINSICS)
+	// Set CPU extension definitions.
+	#if defined(__SSE2__) || _M_IX86_FP >= 2
+		#define DAEMON_CPU_INTRINSICS_i686_sse2
+	#endif
+
+	#if defined(__SSE__) || _M_IX86_FP >= 1
+		#define DAEMON_CPU_INTRINSICS_i686_sse
```

Comment:
Maybe we should call it sse1 for clarity? It might be hard to tell since we don't have any SSE2 code for now.

Reply:
I believe SSE is OK. I don't think it's a good idea to suppose tha “something” means “something and later”, in the same way talking about AVX doesn't assume AVX-512 is available.